src/engine/routing_algorithms/map_matching.cpp

```diff
@@ -198,9 +198,9 @@ SubMatchingList mapMatching(SearchEngineData<Algorithm> &engine_working_data,
 
             const auto haversine_distance = util::coordinate_calculation::haversineDistance(
                 prev_coordinate, current_coordinate);
-            // assumes minumum of 0.1 m/s
-            const int duration_upper_bound =
-                ((haversine_distance + max_distance_delta) * 0.25) * 10;
+            // assumes minumum of 4 m/s
+            const EdgeWeight weight_upper_bound =
+                ((haversine_distance + max_distance_delta) / 4.) * 10;
```

Comment:
Good catch with the multiplier. 👍 

For non-duration based weights this will not work, but in that case I would expect other parts of the map matching to already break (e.g. the parameters for the probability distributions).